plugins/woocommerce/includes/widgets/class-wc-widget-cart.php

```diff
@@ -70,8 +106,17 @@ public function widget( $args, $instance ) {
 			echo '<div class="hide_cart_widget_if_empty">';
 		}
 
-		// Insert cart widget placeholder - code in woocommerce.js will update this on page load.
-		echo '<div class="widget_shopping_cart_content"></div>';
+		global $post;
+		$is_cart_related_block = $this->array_contains( $post->post_content, self::get_blocks() );
+
+		if ( $is_cart_related_block || is_shop() || is_product() || is_product_category() ) {
+			wp_enqueue_script( 'wc-cart-fragments' );
+			// Insert cart widget placeholder - code in woocommerce.js will update this on page load.
+			echo '<div class="widget_shopping_cart_content"></div>';
+		} else {
+			// Insert cart widget with the Mini-cart content.
+			echo '<div class="widget_shopping_cart_content">' . woocommerce_mini_cart() . '</div>';
+		}
```

Comment:
If `wc-cart-fragments` will fetch `woocommerce_mini_cart` over the wire anyway, why not just always print it, and only enqueue the script if needed?

Reply:
@senadir, the `wc-cart-fragments` will fetch `woocommerce_mini_cart` for the first time only. After that, it will get the cart data from the Browser's session storage if the cart content didn't get updated between page loads (see [line](https://github.com/woocommerce/woocommerce/blob/try/load-cart-fragments-script-on-specific-pages/plugins/woocommerce/client/legacy/js/frontend/cart-fragments.js#L144-L148)). I believe there is a performance benefit to not print the cart content on each page load for pages that loads the `wc-cart-fragments` script. Please let me know if you think otherwise! 🙌

Reply:
@Aljullu might have more context on the Mini cart loading.